hw5/task28.py

```diff
@@ -0,0 +1,25 @@
+"""
+Напишите рекурсивную функцию sum(a, b), возвращающую сумму двух целых неотрицательных чисел.
+Из всех арифметических операций допускаются только +1 и -1.
+Также нельзя использовать циклы.
+
+*Пример:*
+
+2 2
+    4
+"""
+
+
+def summ(n1, n2):
+    if n1 == 0 and n2 == 0:
+        return 0
+    elif n1 == 0:
+        return summ(n1, n2 - 1) + 1
+    elif n2 == 0:
+        return summ(n1 - 1, n2) + 1
+    return summ(n1 - 1, n2 - 1) + 1 + 1
```

Comment:
Наверное так будет проще. Избавимся от лишних вызовов функции
```suggestion
    elif n1 == 0:
        return n2
    elif n2 == 0:
        return n1
    return summ(n1 - 1, n2 - 1) + 1 + 1
```

Reply:
@samodurOFF тогда получается, что фактически мы нарушаем требование использовать для складывания только 1, так как при возврате n2/n1 мы впоследствии сделаем сложение с числом больше чем 1

Reply:
Не совсем. Это будет не явно, ведь мы не совершает никакого математического действия, просто возвращаем число.

Reply:
\+